### PR TITLE
update trtllm-gen fp4 autotuner and routing

### PR DESCRIPTION
**Pull request by IwakuraRein**
## 📌 Description

- remove `.to(torch.bfloat16)` which causes illegal memory access when using deepseek v3 routing.
- handle the case when `routing_logits` is `None` to support `trtllm_fp4_block_scale_routed_moe`
- fix the `invalid argument` error in llama and renormalize routing

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [ ] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

### CODE REVIEW

File: flashinfer/fused_moe/core.py [hunk 1]
```diff
@@ -1142,7 +1142,7 @@ def forward(
 
             # TODO(siyuan): support fp8
             moe_op.trtllm_fp4_block_scale_moe(
-                routing_logits.to(torch.bfloat16),
+                routing_logits,
```
Comment (yzh119) [inline flashinfer/fused_moe/core.py:1145]: > remove .to(torch.bfloat16) which causes illegal memory access when using deepseek v3 routing.

could you explain why?

Comment (IwakuraRein) [reply flashinfer/fused_moe/core.py:1145]: @yzh119 Currently DeepSeek V3 requires fp32 routing logits: https://github.com/flashinfer-ai/flashinfer/blob/main/tests/test_trtllm_gen_fused_moe.py#L1987-L1991
